Test/git-issues/git-issue-705.dfy

```diff
@@ -0,0 +1,16 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:cs /spillTargetCode:3 "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:js /spillTargetCode:3 "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:go /spillTargetCode:3 "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:java /spillTargetCode:3 "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype MyDataType = AAA {
+ function method toString(): int { 222 }
+}
+
+method Main() {
+  var dt: MyDataType;
+  print dt, "\n";
```

Comment:
How about additionally printing `dt.toString()` to make sure that it compiles to the user-defined routine that returns `222`?